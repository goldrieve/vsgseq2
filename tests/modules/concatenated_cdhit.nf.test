nextflow_process {

    name "Test Process CONCATENATED_CDHIT"
    script "modules/concatenated_cdhit.nf"
    process "CONCATENATED_CDHIT"

    test("Should run cd-hit on concatenated VSGs") {

        when {
            params {
                cdhit_id = 0.94
                cdhit_as = 0.94
            }
            process {
                """
                input [0] = "$projectDir/tests/data/concatenated_vsgs.fasta"
                input [1] = params.cdhit_id
                input [2] = params.cdhit_as
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
