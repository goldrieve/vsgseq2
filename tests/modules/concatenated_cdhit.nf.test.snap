{
    "Should run cd-hit on concatenated VSGs": {
        "content": [
            {
                "0": [
                    "VSGome.fasta:md5,4eacb0f84c72f87f4fffe227e70bcb7f"
                ],
                "1": [
                    "VSGome.fasta.clstr:md5,e2fe520e2b535152d14fc91f7e6a7760"
                ],
                "clstr": [
                    "VSGome.fasta.clstr:md5,e2fe520e2b535152d14fc91f7e6a7760"
                ],
                "vsgome": [
                    "VSGome.fasta:md5,4eacb0f84c72f87f4fffe227e70bcb7f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-21T15:04:03.56849463"
    }
}